test/Microsoft.AspNetCore.Mvc.Razor.Extensions.Test/TestFiles/Input/RazorPages.cshtml

```diff
@@ -0,0 +1,40 @@
+﻿@page
+
+@model NewModel
+@addTagHelper "*, TestAssembly"
+@using Microsoft.AspNetCore.Mvc.RazorPages
+
+@functions {
+    public class NewModel : PageModel
+    {
+        public IActionResult OnPost(Customer customer)
+        {
+            Name = customer.Name;
+            return Redirect("~/customers/inlinepagemodels/");
+        }
+
+        public string Name { get; set; }
+    }
+
+    public class Customer
+    {
+        public string Name { get; set; }
+    }
+}
+
+<h1>New Customer</h1>
+<form method="post" class="form-horizontal" asp-antiforgery="true">
+    <div asp-validation-summary="All" class="text-danger"></div>
+    <div class="form-group">
+        <label asp-for="Name" class="col-md-2 control-label"></label>
```

Comment:
Nuke the MVCisms (asp-for) from here since you have TagHelpers applying elsewhere and we don't want to cause confusion.